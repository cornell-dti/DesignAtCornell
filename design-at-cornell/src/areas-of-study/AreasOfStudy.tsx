import React, { useEffect, useState } from 'react';
import { PageContainer } from './AreasOfStudyStyles';
import Title from './title/Title';
import Dashboard from './dashboard/Dashboard';
import axios from 'axios';
import { Major } from '../../../server/types';

export type AreaOfStudy = {
  name: string;
  description: string;
};

export type Studies = {
  readonly majors: ReadonlyArray<Major>;
  readonly minors: ReadonlyArray<Major>;
  readonly gradStudies: ReadonlyArray<Major>;
}

const AreasOfStudy = () => {

  const [majors, setMajors] = useState<ReadonlyArray<Major>>([]);

  const majorsURL = 'http://localhost:3000/getMajors';
  useEffect(() => {
    axios
      .get<{
        success: boolean;
        data: Major[];
      }>(majorsURL)
      .then(res => res.data.data)
      .then(setMajors)
  }, []);

  const studies: Studies = {
    majors: majors,
    minors: majors,
    gradStudies: majors
  }

  return (
    <PageContainer>
      <Title />
      <Dashboard {...studies} />
    </PageContainer>
  );
}

export default AreasOfStudy;
