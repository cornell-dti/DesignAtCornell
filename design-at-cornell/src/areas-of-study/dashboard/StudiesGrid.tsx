import React from 'react';
import { Grid, AreaOfStudyButton, AreaOfStudyTag } from '../AreasOfStudyStyles';
import { Major } from '../../../../server/types';
import { Filters } from './Dashboard';

export type StudiesGridProps = {
  readonly studies: ReadonlyArray<Major>;
  readonly filters: Filters;
};

const StudiesGrid = ({ studies, filters }: StudiesGridProps) => (
  <Grid>
    {Object.values(studies).map(({ title, content }) => (
      <AreaOfStudyButton key={title}>
        <h1>{title}</h1>
        <p>{content.type}. {content.designAreas}</p>
        <AreaOfStudyTag highlight={
          filters.schools.has(content.school) ||
          filters.schools.has(content.school.replace('College of ', ''))
        }>
          {content.school}
        </AreaOfStudyTag>
      </AreaOfStudyButton>
    ))}
  </Grid>
);

export default StudiesGrid;
