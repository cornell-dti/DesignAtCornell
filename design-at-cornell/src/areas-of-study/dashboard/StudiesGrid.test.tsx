import React from 'react';
import renderer from 'react-test-renderer';
import StudiesGrid from './StudiesGrid';
import { AreaOfStudy } from '../AreasOfStudy';
import 'jest-styled-components';

describe('<StudiesGrid />', () => {
  const exampleMajor: AreaOfStudy = {
    name: 'test major',
    description: 'This is an example major for testing. ',
  };

  it('renders correctly for no areas of study', () => {
    const areas: AreaOfStudy[] = [];
    const snapshot = renderer.create(<StudiesGrid {...areas} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('renders correctly with one major', () => {
    const areas: AreaOfStudy[] = [exampleMajor];
    const snapshot = renderer.create(<StudiesGrid {...areas} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('renders correctly for 3 majors', () => {
    const areas: AreaOfStudy[] = [
      { ...exampleMajor, name: 'major 1' },
      { ...exampleMajor, name: 'major 2' },
      { ...exampleMajor, name: 'major 3' },
    ];
    const snapshot = renderer.create(<StudiesGrid {...areas} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
