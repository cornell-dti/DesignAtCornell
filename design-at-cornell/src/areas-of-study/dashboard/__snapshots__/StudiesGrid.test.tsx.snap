// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StudiesGrid /> renders correctly for 3 majors 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  width: 287px;
  margin-bottom: 38px;
}

.c1 h1 {
  font-weight: bold;
  font-size: 16px;
  color: black;
  cursor: pointer;
}

.c2 {
  background: white;
  font-size: 13px;
  color: #898989;
}

<div
  className="c0"
>
  <div
    className="c1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <h1>
      Major Title
    </h1>
    <span
      className="c2"
    >
      Engineering. 
    </span>
  </div>
  <div
    className="c1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <h1>
      Major Two
    </h1>
    <span
      className="c2"
    >
      Engineering. 
    </span>
  </div>
  <div
    className="c1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <h1>
      Major Three
    </h1>
    <span
      className="c2"
    >
      Engineering. 
    </span>
  </div>
</div>
`;

exports[`<StudiesGrid /> renders correctly for no areas of study 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div
  className="c0"
/>
`;

exports[`<StudiesGrid /> renders correctly with one major 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  width: 287px;
  margin-bottom: 38px;
}

.c1 h1 {
  font-weight: bold;
  font-size: 16px;
  color: black;
  cursor: pointer;
}

.c2 {
  background: white;
  font-size: 13px;
  color: #898989;
}

<div
  className="c0"
>
  <div
    className="c1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <h1>
      Major Title
    </h1>
    <span
      className="c2"
    >
      Engineering. 
    </span>
  </div>
</div>
`;
